rosbag2_performance/rosbag2_performance_benchmarking/CMakeLists.txt

```diff
@@ -36,25 +36,25 @@ if(BUILD_ROSBAG2_BENCHMARKS)
     src/result_utils.cpp
     src/results_writer.cpp)
 
-  ament_target_dependencies(writer_benchmark
-    rclcpp
-    std_msgs
-    rosbag2_compression
-    rosbag2_cpp
-    rosbag2_storage
-    yaml_cpp_vendor
+  target_link_libraries(writer_benchmark
+    rclcpp::rclcpp
+    ${std_msgs_TARGETS}
+    rosbag2_compression::rosbag2_compression
+    rosbag2_cpp::rosbag2_cpp
+    rosbag2_storage::rosbag2_storage
+    ${YAML_CPP_LIBRARIES}
   )
 
-  ament_target_dependencies(benchmark_publishers
-    rclcpp
-    rosbag2_storage
-    std_msgs
-    yaml_cpp_vendor
+  target_link_libraries(benchmark_publishers
+    rclcpp::rclcpp
+    ${std_msgs_TARGETS}
+\    rosbag2_storage::rosbag2_storage
+    ${YAML_CPP_LIBRARIES}
```

Comment:
Out of curiosity, what was a rational behind decision to replace `yaml_cpp_vendor` by `${YAML_CPP_LIBRARIES}`?

Reply:
Yeah, we shouldn't use this.  I think this should be `yaml-cpp::yaml-cpp`.

Reply:
@MichaelOrlov @clalancette I replaced it with exported target. [here](https://github.com/ros2/rosbag2/pull/1202/commits/7c8e43ac4e71095d1400f1774b75380fd40902fc)

Reply:
@wep21 Could you please make the same replacement in all over other places in your original changes?  
It's almost in every CMake file.

Reply:
@wep21 Thanks for follow up on my comments.  
Could you please explain rational behind replacing `yaml-cpp::yaml-cpp` to just simple `yaml-cpp`?  
I am curious why it's different from others.

Reply:
@MichaelOrlov Please see [this comment](https://github.com/ros2/rosbag2/pull/1202#discussion_r1051421244).

Reply:
Ok got it.